pkg/gofr/migration/mongo.go

```diff
@@ -0,0 +1,100 @@
+package migration
+
+import (
+	"context"
+	"time"
+
+	"gofr.dev/pkg/gofr/container"
+)
+
+type mongoDS struct {
+	container.Mongo
+}
+
+type mongoMigrator struct {
+	container.Mongo
+	migrator
+}
+
+// apply initializes mongoMigrator using the Mongo interface.
+func (ds mongoDS) apply(m migrator) migrator {
+	return mongoMigrator{
+		Mongo:    ds.Mongo,
+		migrator: m,
+	}
+}
+
+const (
+	mongoMigrationCollection = "gofr_migrations"
+)
+
+// checkAndCreateMigrationTable initializes a MongoDB collection if it doesn't exist.
+func (mg mongoMigrator) checkAndCreateMigrationTable(_ *container.Container) error {
```

Comment:
I would pass the context instead of using context.Background

Reply:
You are absolutely right passing context is a better option but it can be a breaking change for all other datasource migrators as our migrator interface doesn't have context as paramter in the following methods:

```go
type migrator interface {
	checkAndCreateMigrationTable(c *container.Container) error
	getLastMigration(c *container.Container) int64

	beginTransaction(c *container.Container) transactionData

	commitMigration(c *container.Container, data transactionData) error
	rollback(c *container.Container, data transactionData)
}
```


So now should we do it for all the migrators in this PR or open a new issue to change all of them after this gets merged?

Reply:
We will be handling this issue in a new PR then.

Reply:
I opened #1183

- #1183